golang/networks/network_context.go

```diff
@@ -172,41 +160,55 @@ func (networkCtx *NetworkContext) AddServiceToPartition(
 	}
 	resp, err := networkCtx.client.StartService(ctx, startServiceArgs)
 	if err != nil {
-		return nil, nil, nil, stacktrace.Propagate(err, "An error occurred starting the service with the Kurtosis API")
+		return nil, nil, stacktrace.Propagate(err, "An error occurred starting the service with the Kurtosis API")
 	}
 	logrus.Tracef("Successfully started service with Kurtosis API")
 
-	logrus.Tracef("Creating service interface...")
-	service := containerCreationConfig.GetServiceCreatingFunc()(serviceContext)
-	logrus.Tracef("Successfully created service interface")
-
-	networkCtx.services[serviceId] = service
-
-	availabilityChecker := services.NewDefaultAvailabilityChecker(serviceId, service)
-
-	return service, resp.UsedPortsHostPortBindings, availabilityChecker, nil
+	return serviceContext, resp.UsedPortsHostPortBindings, nil
 }
 
 // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
-func (networkCtx *NetworkContext) GetService(serviceId services.ServiceID) (services.Service, error) {
-	networkCtx.mutex.Lock()
-	defer networkCtx.mutex.Unlock()
+func (networkCtx *NetworkContext) GetServiceContext(serviceId services.ServiceID) (*services.ServiceContext, error) {
+	getServiceInfoArgs := &core_api_bindings.GetServiceInfoArgs{
+		ServiceId: string(serviceId),
+	}
+	serviceResponse, err := networkCtx.client.GetServiceInfo(context.Background(), getServiceInfoArgs)
+	if err != nil {
+		return nil, stacktrace.Propagate(
+			err,
+			"An error occurred when trying to get info for service '%v'",
+			serviceId)
+	}
 
-	service, found := networkCtx.services[serviceId]
-	if !found {
-		return nil, stacktrace.NewError("No service info found for ID '%v'", serviceId)
+	if serviceResponse.GetIpAddr() == "" {
+		return nil, stacktrace.NewError(
+			"An error occurred when trying to get the IP address from service '%v'",
```

Comment:
Would really recommend flagging in the error message that if this condition is tripped, it's definitely a bug in Kurtosis - we should never get an empty IP address from Kurt API